client/src/pages/organizations/Show.js

```diff
@@ -315,9 +315,19 @@ const OrganizationShow = ({ pageDispatchers }) => {
                           modelType="Organization"
                           model={organization.parentOrg}
                         >
-                          {organization.parentOrg.shortName}{" "}
-                          {organization.parentOrg.longName}{" "}
-                          {organization.parentOrg.identificationCode}
+                          {isPrincipalOrg
+                            ? `${organization.parentOrg.shortName} ${
+                              organization.parentOrg.identificationCode
+                                ? ` \\ UIC: ${organization.parentOrg.identificationCode}`
+                                : " \\ UIC: Not specified."
+                            }`
+                            : `${organization.parentOrg.shortName} ${
+                              organization.parentOrg.longName
+                            } ${
+                              organization.parentOrg.identificationCode
+                                ? `${organization.parentOrg.identificationCode}`
+                                : ""
+                            }`}
```

Comment:
This code will now appear in three places (a similar change needs to be made in `organizations/Form.js` as well), so it warrants a separate utility function. Also, `UIC` is now hardcoded, but it needs to be looked up in the dictionary.
I propose the following change instead:
```
diff --git a/client/src/models/Organization.js b/client/src/models/Organization.js
index 2872bc178..6ea6aa6e5 100644
--- a/client/src/models/Organization.js
+++ b/client/src/models/Organization.js
@@ -145,4 +145,14 @@ export default class Organization extends Model {
   toString() {
     return this.shortName || this.longName || this.identificationCode
   }
+
+  static toIdentificationCodeString(organization) {
+    return organization.type === Organization.TYPE.PRINCIPAL_ORG
+      ? `${organization.shortName} \\ ${
+        Settings.fields.advisor.org.identificationCode.label
+      }: ${organization.identificationCode || "Not specified"}`
+      : `${organization.shortName} ${organization.longName} ${
+        organization.identificationCode || ""
+      }`
+  }
 }
diff --git a/client/src/pages/organizations/Form.js b/client/src/pages/organizations/Form.js
index 04a655a9e..912e37669 100644
--- a/client/src/pages/organizations/Form.js
+++ b/client/src/pages/organizations/Form.js
@@ -193,9 +193,9 @@ const OrganizationForm = ({ edit, title, initialValues }) => {
                             modelType="Organization"
                             model={values.parentOrg}
                           >
-                            {values.parentOrg.shortName}{" "}
-                            {values.parentOrg.longName}{" "}
-                            {values.parentOrg.identificationCode}
+                            {Organization.toIdentificationCodeString(
+                              values.parentOrg
+                            )}
                           </LinkTo>
                         )
                       }
diff --git a/client/src/pages/organizations/Show.js b/client/src/pages/organizations/Show.js
index ea08c381a..84d2bba26 100644
--- a/client/src/pages/organizations/Show.js
+++ b/client/src/pages/organizations/Show.js
@@ -56,12 +56,14 @@ const GQL_GET_ORGANIZATION = gql`
         shortName
         longName
         identificationCode
+        type
       }
       childrenOrgs(query: { pageNum: 0, pageSize: 0, status: ACTIVE }) {
         uuid
         shortName
         longName
         identificationCode
+        type
       }
       positions {
         uuid
@@ -315,19 +317,9 @@ const OrganizationShow = ({ pageDispatchers }) => {
                           modelType="Organization"
                           model={organization.parentOrg}
                         >
-                          {isPrincipalOrg
-                            ? `${organization.parentOrg.shortName} ${
-                              organization.parentOrg.identificationCode
-                                ? ` \\ UIC: ${organization.parentOrg.identificationCode}`
-                                : " \\ UIC: Not specified."
-                            }`
-                            : `${organization.parentOrg.shortName} ${
-                              organization.parentOrg.longName
-                            } ${
-                              organization.parentOrg.identificationCode
-                                ? `${organization.parentOrg.identificationCode}`
-                                : ""
-                            }`}
+                          {Organization.toIdentificationCodeString(
+                            organization.parentOrg
+                          )}
                         </LinkTo>
                       )
                     }
@@ -380,19 +372,9 @@ const OrganizationShow = ({ pageDispatchers }) => {
                                 modelType="Organization"
                                 model={organization}
                               >
-                                {isPrincipalOrg
-                                  ? `${organization.shortName} ${
-                                    organization.identificationCode
-                                      ? ` \\ UIC: ${organization.identificationCode}`
-                                      : " \\ UIC: Not specified."
-                                  }`
-                                  : `${organization.shortName} ${
-                                    organization.longName
-                                  } ${
-                                    organization.identificationCode
-                                      ? `${organization.identificationCode}`
-                                      : ""
-                                  }`}
+                                {Organization.toIdentificationCodeString(
+                                  organization
+                                )}
                               </LinkTo>
                             </ListGroupItem>
                           ))}
```